tests/rptest/tests/raft_availability_test.py

```diff
@@ -421,9 +455,25 @@ def test_follower_isolation(self):
                 FailureSpec(FailureSpec.FAILURE_ISOLATE,
                             self.redpanda.get_node(replicas[1])))
 
-            # expect messages to be produced and consumed without a timeout
-            for i in range(0, 128):
-                self._ping_pong()
+            nodes = self.redpanda.get_nodes()
+            nodes = [node for node in nodes if node.id != replicas[1]]
+            admin.wait_stable_leader("id_allocator",
+                                     namespace="kafka_internal",
+                                     replication=3,
+                                     timeout_s=ELECTION_TIMEOUT * 2,
+                                     nodes=nodes,
+                                     check=lambda node: node.id != replicas[1])
+
+            count = 0
+            while count < 128:
+                # a request may fail when a init producer request lands on
+                # the isolated node (a client picks an initial node at random)
+                # retrying a request to mitigate this problem
+                wait_until(lambda: self._is_available(1) is True,
+                           timeout_sec=ELECTION_TIMEOUT * 6,
+                           backoff_sec=0.5,
+                           err_msg=f"Cluster isn't available")
+                count += 1
```

Comment:
the commit message for this patch:

> ducky: adjust raft_availability for idempotency

IMO doesn't explain what this patch is doing and it isn't obvious from reading the code. even a few sentences is really helpful later on when looking back at code.

Reply:
done

Reply:
Thank you very much :)